controllers/devworkspace/solver/che_routing_external_tls_config_test.go

```diff
@@ -0,0 +1,182 @@
+//
+// Copyright (c) 2019-2025 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package solver
+
+import (
+	"testing"
+
+	"github.com/devfile/devworkspace-operator/pkg/infrastructure"
+	chev2 "github.com/eclipse-che/che-operator/api/v2"
+	controller "github.com/eclipse-che/che-operator/controllers/devworkspace"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func TestExternalTLSConfigForIngresses(t *testing.T) {
+	infrastructure.InitializeForTesting(infrastructure.Kubernetes)
+
+	mgr := &chev2.CheCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       "che",
+			Namespace:  "ns",
+			Finalizers: []string{controller.FinalizerName},
+		},
+		Spec: chev2.CheClusterSpec{
+			Networking: chev2.CheClusterSpecNetworking{
+				TlsSecretName: "tlsSecret",
+				Hostname:      "beyond.comprehension",
+				Domain:        "almost.trivial",
+				Annotations: map[string]string{
+					"default_annotation_key": "default_annotation_value",
+				},
+			},
+			DevEnvironments: chev2.CheClusterDevEnvironments{
+				Networking: &chev2.DevEnvironmentNetworking{
+					ExternalTLSConfig: &chev2.ExternalTLSConfig{
+						Enabled: pointer.Bool(true),
+						Annotations: map[string]string{
+							"annotation_key": "annotation_value",
+						},
+						Labels: map[string]string{
+							"label_key": "label_value",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, _, objs := getSpecObjectsForManager(t, mgr, subdomainDevWorkspaceRouting(), userProfileSecret("username"),
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tlsSecret",
+				Namespace: "ns",
+			},
+			Data: map[string][]byte{
+				"tls.key": []byte("asdf"),
+				"tls.crt": []byte("qwer"),
+			},
+		})
+
+	assert.Equal(t, 3, len(objs.Ingresses))
```

Comment:
nit,  Maybe we can replace `assert` with [`require`](https://github.com/stretchr/testify#require-package) for critical assertions here so that we fail fast.
```suggestion
	require.Equal(t, 3, len(objs.Ingresses))
```